include/mathfu/vector_2.h

```diff
@@ -162,6 +162,19 @@ class Vector<float, 2> {
     return *this;
   }
 
+  inline bool operator==(const Vector<float, 2>& v) const {
+    for (int i = 0; i < 2; ++i) {
+      if (this->data_.float_array[i] != v[i]) {
```

Comment:
It's probably be cleaner to do

for (int i = 0; i < 2; ++i) {
  if ((*this)[i] != v[i]) return false;
}
return true;

instead so there are less places to clean up if we change the internal representation of the vector in future.